src/liballoc/vec.rs

```diff
@@ -2379,7 +2379,7 @@ unsafe impl<#[may_dangle] T> Drop for Vec<T> {
     fn drop(&mut self) {
         unsafe {
             // use drop for [T]
-            ptr::drop_in_place(&mut self[..]);
+            ptr::drop_in_place(ptr::slice_from_raw_parts_mut(self.as_mut_ptr(), self.len))
```

Comment:
Might be worth adding a brief comment for why we use this complicated operation here instead of the simple one.

Really, already for https://github.com/rust-lang/rust/pull/70558, what we should really have is a way to get a raw slice for the entire `Vec`, and then subslicing on raw slices. Then we would not have to call `slice_from_raw_parts` for any of the methods changed in that PR or here.

Reply:
Comment added, though it's very vague since I don't dare say so much about what it actually allows